core/src/test/java/nl/nn/adapterframework/jdbc/StoredProcedureQuerySenderTest.java

```diff
@@ -246,6 +276,40 @@ public void testStoredProcedureReturningResultSet() throws Exception {
 		assertXmlEquals(sb.toString(), actual);
 	}
 
+	@Test
+	public void testCallFunction() throws Exception {
+		assumeThat("CALL to custom function only tested on Oracle so far", productKey, equalToIgnoringCase("Oracle"));
+
+		// Arrange
+		sender.setQuery("{ ? = call add_numbers(?, ?) }");
+		sender.setQueryType(JdbcQuerySenderBase.QueryType.OTHER.name());
+		sender.setOutputParameters("1");
+		sender.setScalar(true);
+
+		Parameter dummyParam = new Parameter("dummy", "0");
+		dummyParam.setType(Parameter.ParameterType.INTEGER);
+		dummyParam.configure();
+		sender.addParameter(dummyParam);
+		Parameter p1 = new Parameter("one", "1");
+		p1.setType(Parameter.ParameterType.INTEGER);
+		p1.configure();
+		sender.addParameter(p1);
+		Parameter p2 = new Parameter("two", "2");
+		p2.setType(Parameter.ParameterType.INTEGER);
+		p2.configure();
+		sender.addParameter(p2);
```

Comment:
Waarom heb je hier nu 3 parameters?

Reply:
De output is ook een parameter, de dummy-parameter als eerste. Omdat vanwege de syntax van Oracle die vooraan staat, moet die als `Parameter` worden gedefinieerd. Anders lukt het me niet om de andere 2 op hun plek te krijgen, en komen er exceptions dat niet alle parameters gevuld zijn.

Bonus: door nu het type te gebruiken van de Parameter hoeft het niet te worden gespecificeerd in het `outputParameters` attribuut.

(En deze test-case zit erin omdat Ali een Oracle Function heeft die vergelijkbaar is).

Reply:
Dan is het handiger om altijd alle parameters (in als uit) op te geven, en dan een attribuut waar je de param namen op moet geven die output params zijn?

Reply:
Dat zou ook kunnen ja, dat `outputParameters` de namen geeft ipv posities.

Ik zit nog even met 1 ding, en dat is de parameter-types die niet altijd goed mappen naar JDBC types. Als een stored procedure een CURSOR of REF_CURSOR terug geeft in een OUT-parameter, dan krijg je in die parameter een hele result-set.
Er is nu geen parameter type waarmee dat kan worden aangegeven. 

Zorg voor later wellicht.